halo2_proofs/src/dev/metadata.rs

```diff
@@ -33,6 +36,42 @@ impl From<plonk::Column<Any>> for Column {
     }
 }
 
+/// A helper structure that allows to print a Column with it's annotation as a single structure.
+#[derive(Debug, Clone)]
+struct DebugColumn {
+    /// The type of the column.
+    column_type: Any,
+    /// The index of the column.
+    index: usize,
+    /// Annotation of the column
+    annotation: String,
+}
```

Comment:
I'll not lie I've thought about this! So there are two things why I went this way:
1. This was the less invasive way on which I could also improve the `VerifyFailure` error display. Otherwise, I couldn't print the annotations within the column itself.

2. The type will no longer be `Copy`. And that (for some quick observations done) leads to a "significant" amount of `.clone()` in the codebase. I haven't checked anything further introducing the attribute. So some other complexities could be hidden.

I can go into that direction anyway and try to implement it if you want to check how it'll look like. But maybe it's better to do it in a separated PR so that the diff leaves it more clear.

Just LMK what you think!